public/controllers/agent/agents.js

```diff
@@ -167,29 +167,62 @@ export class AgentsController {
     this.$scope.isString = item => typeof item === 'string';
     this.$scope.hasSize = obj =>
       obj && typeof obj === 'object' && Object.keys(obj).length;
-    this.$scope.switchConfigTab = (configurationTab, sections) =>
+    this.$scope.switchConfigTab = (configurationTab, sections, navigate = true) => {
+      this.$scope.navigate = navigate;
+      this.$scope.configSubTab = JSON.stringify({ 'configurationTab': configurationTab, 'sections': sections });
+      if (!this.$location.search().configSubTab) {
+        sessionStorage.setItem("configSubTab", this.$scope.configSubTab);
+        this.$location.search('configSubTab', true);
+      }
       this.configurationHandler.switchConfigTab(
         configurationTab,
         sections,
         this.$scope,
         this.$scope.agent.id
       );
-    this.$scope.switchWodle = wodleName =>
+    }
+    this.$scope.switchWodle = (wodleName, navigate = true) => {
+      this.$scope.navigate = navigate;
+      this.$scope.configWodle = wodleName;
+      if (!this.$location.search().configWodle) {
+        this.$location.search('configWodle', this.$scope.configWodle);
+      }
       this.configurationHandler.switchWodle(
         wodleName,
         this.$scope,
         this.$scope.agent.id
-      );
-    this.$scope.switchConfigurationTab = configurationTab =>
+      )
+    };
+    this.$scope.switchConfigurationTab = (configurationTab, navigate) => {
+      this.$scope.navigate = navigate;
       this.configurationHandler.switchConfigurationTab(
         configurationTab,
         this.$scope
       );
-    this.$scope.switchConfigurationSubTab = configurationSubTab =>
+      if (!this.$scope.navigate) {
+        const configSubTab = this.$location.search().configSubTab;
+        if (configSubTab) {
+          const config = sessionStorage.getItem("configSubTab");
```

Comment:
Use this one https://github.com/wazuh/wazuh-kibana-app/blob/master/public/services/app-state.js (move your logic there)

Reply:
Also, clear the storage when it's no longer needed